src/core/shutdown.c

```diff
@@ -159,6 +164,102 @@ static int switch_root_initramfs(void) {
         return switch_root("/run/initramfs", "/oldroot", false, MS_BIND);
 }
 
+/* Read the following fields from /proc/meminfo:
+ *
+ *  NFS_Unstable
+ *  Writeback
+ *  Dirty
+ *
+ * Return true if the sum of these fields is greater than the previous
+ * value input. For all other issues, report the failure and indicate that
+ * the sync is not making progress.
+ */
+static bool sync_making_progress(unsigned long long *prev_dirty) {
+        _cleanup_fclose_ FILE *f = NULL;
+        char line[LINE_MAX];
+        int r = false;
+        unsigned long long val = 0;
+
+        f = fopen("/proc/meminfo", "re");
+        if (!f) {
+                return log_warning_errno(errno, "Failed to open /proc/meminfo: %m");
+        }
```

Comment:
no {} around single line if blocks, see CODING_STYLE

Reply:
Sorry about that. Artifact from previous debug statement. Fixed in next version.